bind/gen.go

```diff
@@ -0,0 +1,666 @@
+// Copyright 2019 The go-python Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package bind
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// this version uses pybindgen and a generated .go file to do the binding
+
+// for all preambles: 1 = name of package (outname), 2 = cmdstr
+
+// GoHandle is the type to use for the Handle map key, go-side
+// could be a string for more informative but slower handles
+var GoHandle = "int64"
+var CGoHandle = "C.longlong"
+var PyHandle = "int64_t"
+
+// var GoHandle = "string"
+// var CGoHandle = "*C.char"
+// var PyHandle = "char*"
+
+// 3 = libcfg, 4 = GoHandle, 5 = CGoHandle, 6 = all imports, 7 = mainstr, 8 = exe pre C, 9 = exe pre go
+const (
+	goPreamble = `/*
+cgo stubs for package %[1]s.
+File is generated by gopy. Do not edit.
+%[2]s
+*/
+
+package main
+
+/*
+#cgo pkg-config: %[3]s
+// #define Py_LIMITED_API // need full API for PyRun*
+#include <Python.h>
+typedef uint8_t bool;
+// static inline is trick for avoiding need for extra .c file
+// the following are used for build value -- switch on reflect.Kind
+// or the types equivalent
+static inline PyObject* gopy_build_bool(uint8_t val) {
+	return Py_BuildValue("b", val);
+}
+static inline PyObject* gopy_build_int64(int64_t val) {
+	return Py_BuildValue("k", val);
+}
+static inline PyObject* gopy_build_uint64(uint64_t val) {
+	return Py_BuildValue("K", val);
+}
+static inline PyObject* gopy_build_float64(double val) {
+	return Py_BuildValue("d", val);
+}
+static inline PyObject* gopy_build_string(const char* val) {
+	return Py_BuildValue("s", val);
+}
+static inline void gopy_decref(PyObject* obj) { // macro
+	Py_XDECREF(obj);
+}
+static inline void gopy_incref(PyObject* obj) { // macro
+	Py_XINCREF(obj);
+}
+static inline int gopy_method_check(PyObject* obj) { // macro
+	return PyMethod_Check(obj);
+}
+static inline void gopy_err_handle() {
+	if(PyErr_Occurred() != NULL) {
+		PyErr_Print();
+	}
+}
+%[8]s
+*/
+import "C"
+import (
+	"github.com/goki/gopy/gopyh" // handler
```

Comment:
this would need to be `go-python/gopy` right?

Reply:
after the PR is accepted, yes, but until then, wouldn't be able to use / test this forked version.. is there any good general soln to this problem?